### PR TITLE
contact detail component

### PR DESCRIPTION
**Pull request by francellanjones**
Adds a custom component for contact details. 
Comes with:
   - prepopulated dropdown
   - selected number in the case of a TVH preferred number
   - option to add a new phone number
   - validation on inputted phone number

To test you can go to the example.html page to see it in action
Dropdown
custom component dropdown
custom component populated dropdown
Freetext
custom component freetext

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,17 +1,28 @@
 {
   "name": "@mtvh/mtvh-formio",
-  "version": "2.0.12",
+  "version": "2.0.10",
```
Comment (robbiepaul) [inline package.json:3]: Should we bump the version?
```suggestion
  "version": "2.1.0",
```

File: src/components/mtvhContactDetail/mtvhContactDetail.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  mtvhValid(){
+    this.refs.inlineValidation.innerHTML = '<b style="color:green">Valid UK phone number</b>';
+  }
+
+  isPhoneNumberValid(input){
+    return /^(?:\W*\d){11}\W*$/.test(input)
+  }
+
+  inputtedPhoneNumber(){
+    return this.refs.newDetailsInput.value.replace(/\W+/g, '');
```
Comment (robbiepaul) [inline src/components/mtvhContactDetail/mtvhContactDetail.ts:188]: Nice 👍

File: src/components/mtvhContactDetail/mtvhContactDetail.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  /*
+  getValue() {
+    return super.getValue();
+  }
+
+  updateValue(value, flags = {}) {
+    return super.updateValue(value, flags);
+  }
+  */
+
```
Comment (robbiepaul) [inline src/components/mtvhContactDetail/mtvhContactDetail.ts:244]: Let's remove these if they aren't required

```suggestion
```

File: src/templates/bootstrap/mtvhContactDetail/form.ejs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      <span class="input-switch-icon"></span> Provide a different phone number
+      </a>
+    </div>
+  </div>
+
+  <div ref="manualInput">
+    <input ref="newDetailsInput" type='text' class="form-control  mb-1" autocomplete="tel">
+    <div ref="inlineValidation"></div>
+    <div class="mt-2">
+      <a ref="switchToDropdown", href="#">
```
Comment (robbiepaul) [inline src/templates/bootstrap/mtvhContactDetail/form.ejs:20]:
```suggestion
      <a ref="switchToDropdown" href="#">
```

File: src/templates/bootstrap/mtvhContactDetail/form.ejs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{% if (ctx.readOnly === true) { %}
+<input value="{% if (ctx.value) { %}{{ctx.value}}{% } else { %}{% } %}" spellcheck="true" lang="en" class="form-control" type="input" name="data[{{ctx.component.key}}]" ref="mtvhContact" readonly>
+{% } else { %}
+<div>
+  <div ref="selectInput">
+    <select lang="en" class="form-control mb-1 formio-component-mtvhContactDetail" type="text" id="{{ctx.instance.id}}-{{ctx.component.key}}-selectAddress" ref="existingDetailsDropdown">
+      <option value="" selected="selected"></option>
+    </select>
+    <div class="mt-2">
+      <a ref="switchToFreetext", href="#">
```
Comment (robbiepaul) [inline src/templates/bootstrap/mtvhContactDetail/form.ejs:10]:
```suggestion
      <a ref="switchToFreetext" href="#">
```

File: fonts/fonts.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+@font-face{
+    font-family:"Paralucent W05"; font-weight:600; font-style: normal;
+    src:url("./16dbb7d2-6ea2-4f39-b5b4-93be0cf7c738.woff2") format("woff2"),url("./084495f6-ff55-483d-b05c-36ce1704d47c.woff") format("woff");
+}
```
Comment (robbiepaul) [inline fonts/fonts.css:4]: Are these font changes needed?

**Review by jmc-tv — CHANGES_REQUESTED** [review]
Have the free-text entry of the email address make use of the already existing email typo suggestor we use elsewhere when users or agents enter an email.

**Comment by njall-mtvh** [comment]
This isn't an email component @jmc-tv - it's only a phone number component. 

We are only doing the phone number component here because we decided that the email would: 
a) be just an acknowledgement that an SRQ has been raised to the default email, with no submission information attached so as not to risk GDPR issues
b) therefore not be necessary for the minimum viable product to add an email

**Review by jmc-tv — DISMISSED** [review]

**Comment by jmc-tv** [comment]
> This isn't an email component @jmc-tv - it's only a phone number component.
> 
> We are only doing the phone number component here because we decided that the email would: a) be just an acknowledgement that an SRQ has been raised to the default email, with no submission information attached so as not to risk GDPR issues b) therefore not be necessary for the minimum viable product to add an email

Grand so.

File: fonts/fonts.css (hunk 6 above)
Comment (njall-mtvh) [reply fonts/fonts.css:4]: 🤷 https://github.com/thames-valley-housing/mtvh-formio/commits/e05b1586052a95a1ffbbbe2cd3dbf2f2126900cd/fonts/fonts.css

Comment (njall-mtvh) [reply fonts/fonts.css:4]: Matts added it to this branch, but I wouldn't expect he'd want it to go live given the commit is `playing`. I'll remove it from this PR and let Matt know. Good spot!

Comment (njall-mtvh) [reply fonts/fonts.css:4]: I dunno whether to remove it tbh. Can't tell if it'll be a problem or not... Any thoughts?

**Review by robbiepaul — APPROVED** [review]

**Pull request merged**